Unexpected output of compiling/linting a smart contract w/o a validator function

**Describe the bug**
Compiling a smart contract without a `validator` function outputs an unwanted output:
```
Traceback (most recent call last):
  File "/Users/chrissi/Library/Caches/pypoetry/virtualenvs/opshin-_DC4-49p-py3.11/bin/opshin", line 6, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/chrissi/Desktop/repos/cardano/OpShin Toolchain/public repo/opshin/opshin/__main__.py", line 193, in main
    annotations = list(sc.validator.__annotations__.values())
                       ^^^^^^^^^^^^
AttributeError: module '__tmp_opshin' has no attribute 'validator'
```

The problem is that in line 119 of `opshin/__main__.py` it is already expected that the smart contract has a function called `validator`.

Whereas this would normally be checked at a later step when the source_code gets parsed/compiled.
Then the error would be normally get caught by an assertion in the method `visit_Module` in `opshin/compiler.py`.

This impacts also the linting functionality since this is an uncaught exception and then it doesn't get printed out correctly.


**To Reproduce**
Steps to reproduce the behavior:
1. create a python file without an `validator()` function
2. run something like `opshin build any broken.py`
3. See unwanted output

**Expected behavior**
expected should be something like:
```
Traceback (most recent call last):
  File "/Users/chrissi/Library/Caches/pypoetry/virtualenvs/opshin-_DC4-49p-py3.11/bin/opshin", line 6, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/chrissi/Desktop/repos/cardano/OpShin Toolchain/public repo/opshin/opshin/__main__.py", line 161, in main
    raise SyntaxError(
  File "examples/broken.py", line 1
    from opshin.prelude import *
AssertionError: Could not find function named validator
Note that opshin errors may be overly restrictive as they aim to prevent code with unintended consequences.
```
---
current version of the main branch with commit `9e3ad8a8c6c3d7d55262a43eb5b5b35939d33695`

--- a/opshin/__main__.py
+++ b/opshin/__main__.py
@@ -116,6 +116,12 @@ def check_params(
     num_onchain_params = 3 if purpose == Purpose.spending or force_three_params else 2
     onchain_params = validator_args[-1 - num_onchain_params : -1]
     param_types = validator_args[: -1 - num_onchain_params]
+    required_onchain_parameters = 3 if purpose == Purpose.spending else 2
+    assert (
+        len(onchain_params) == required_onchain_parameters
+    ), f"""\
+{purpose.value.capitalize()} validator must expect {required_onchain_parameters} parameters at evaluation (on-chain), but was specified to have {len(onchain_params)}.
+Make sure the validator expects parameters {'datum, ' if purpose == Purpose.spending else ''}redeemer and script context."""
 
     if command in (Command.eval, Command.eval_uplc):
         assert len(validator_params) == len(param_types) + len(

--- a/opshin/__main__.py
+++ b/opshin/__main__.py
@@ -213,7 +213,12 @@ def main():
         assert not args.args, "Can not pass arguments to a library"
         parsed_params = []
     else:
-        annotations = list(sc.validator.__annotations__.values())
+        try:
+            annotations = list(sc.validator.__annotations__.values())
+        except AttributeError:
+            raise AssertionError(
+                f"Contract has no function called 'validator'. Make sure the compiled contract contains one function called 'validator' or {command.value} using `opshin {command.value} lib {str(input_file)}`."
+            )
         if "return" not in sc.validator.__annotations__:
             annotations.append(prelude.Anything)
         parsed_params = []

